### DISCUSSION TITLE
Will Bromite maintain support for Android 6?

### DISCUSSION THREAD

**Discussion by Rikk** [bromite/bromite#2413] [General]
Will Bromite maintain support for Android 6 when it get updated to chromium 107, or will it follow Chrome and forget Android 6 users?

**Comment by n-buna404** [comment]
Bromite followed upstream from previous experience. But u may wait for official response from project maintiner.

**Comment by n-buna404** [comment]
Ram problem can be mitigated by using armv7a version. If chrome upstream
suddenly dropped armv7a even android 9 with 4gb ram will succumb to
performance issue. Just to get back to the topic, I think bromite will most
likely following upstream.

**Comment by csagan5** [comment]
There are multiple factors at play:
* upstream usually drops an older version to be able to build with a more recent SDK, and that usually also means better security and performance
* once upstream drops an older version the patches needed to keep building for the older version might become too complex/unsustainable

It has to be evaluated each time whether it's possible to re-introduce support of an older version or not.

**Reply by Rikk** [reply]
Chrome is already built using targetSDK 33 for quite some time. What I think is Google dropped Android 6 just due to politics, not because an incompatibilty caused by a change or feature.

Here Chrome (yes, armv7a) runs more stable than other chromium browsers or Webview-based ones (sorry, not tried Bromite yet). I would consider moving to Bromite, if it keeps supporting us.

I don't understand why Play Services is still updated for Android 4.4+ while Internet security is requiring Android 6+ now.

**Reply by csagan5** [reply]
> What I think is Google dropped Android 6 just due to politics, not because an incompatibility caused by a change or feature.

Can you prove this statement? Have you checked the Chromium commits/issues to see if there is any explanation about the change?

**Reply by Rikk** [reply]
Only a dev specialized in chromium code can confirm it, not me.

**Reply by csagan5** [reply]
> Only a dev specialized in chromium code can confirm it, not me.

Then it is not due to politics, there is no evidence of this.

**Reply by Rikk** [reply]
Google is used to stop supporting some Android versions after some time, not because of incompatible changes (although dropping support allows implementating features otherwise incompatible in future versions). I used the term "politics", you may use another you think fits better. Not only Chrome, but several other apps already made this transition. I would like 1) Google published with anticipation its plans to unsupport Android versions and 2) Chrome would be the last app discontinued for Android 6.

**Reply by uazo** [reply]
> Why?
> We value our users and want to keep them safe on the web, but maintaining Chrome on operating system versions that are no longer supported reduces our ability to move the rest of Chrome forward. As other apps have done, we have decided to end support for Android 6.0 Marshmallow now that [almost 7 years have passed since its final update](https://developer.android.com/studio/releases/platforms).
> 
> When?
> Chrome M106 is the last release that will support Android 6.0 Marshmallow. You can find the [Chrome release schedule here](https://chromiumdash.appspot.com/schedule).
> Chrome M107 will not be shipped to Android 6.0 Marshmallow so you can now make cleanups for this on the main branch.

source https://groups.google.com/a/chromium.org/g/chromium-dev/c/z_RvoPoIeoM/m/eBsuNkvIAQAJ
this is the [list of reverts](https://bugs.chromium.org/p/chromium/issues/detail?id=1363002) to make and keep, impossible job

**Reply by csagan5** [reply]
In this case I cannot see a compelling reason to drop support, all those changes put together are really not a lot and nowhere it is explained what pushed this deprecation. There is a linked blocking issue but it's not public.

**Reply by uazo** [reply]
>  all those changes put together are really not a lot

maybe i don't understand, but my guess is that a revert of all the linked bugids needs to be done... which then need to be maintained...
do you have something that extracts from the page all the reverts automatically?

**Reply by Rikk** [reply]
> this is the [list of reverts](https://bugs.chromium.org/p/chromium/issues/detail?id=1363002) to make and keep, impossible job

Nice links, thanks. Briefly looking, for now it seems they are only doing cleanups, changing min/maxSdk, removing tests and adjusting build machines.

"reduces our ability to move the rest of Chrome forward" - pretending they have plans for new features, something I don't see in a long time (at least in the UI of Android Chrome).

**Reply by csagan5** [reply]
> my guess is that a revert of all the linked bugids needs to be done... which then need to be maintained...

They are not a lot of changes for the teams behind Chromium. I don't see what type of problem it could create to leave support of Android 6, but we do not have access to the blocked issue so nothing can be concluded.
